docs/src/model_setup/boundary_conditions.md

```diff
@@ -322,3 +424,61 @@ julia> model.tracers.c
 Notice that the specified non-default boundary conditions have been applied at
 top and bottom of both `model.velocities.u` and `model.tracers.c`.
 
+## Immersed boundary conditions
+
+Immersed boundary conditions are supported experimentally. A no-slip boundary condition is specified
+by writing
+
+```jldoctest
+julia> underlying_grid = RectilinearGrid(size=(32, 32, 16), x=(-3, 3), y=(-3, 3), z=(0, 1), topology=(Periodic, Periodic, Bounded));
+
+julia> hill(x, y) = 0.1 + 0.1 * exp(-x^2 - y^2)
+hill (generic function with 1 method)
+
+julia> grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(hill))
+32×32×16 ImmersedBoundaryGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo:
+├── immersed_boundary: GridFittedBottom{OffsetArrays.OffsetMatrix{Float64, Matrix{Float64}}}
+├── underlying_grid: 32×32×16 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── Periodic x ∈ [-3.0, 3.0) regularly spaced with Δx=0.1875
+├── Periodic y ∈ [-3.0, 3.0) regularly spaced with Δy=0.1875
+└── Bounded  z ∈ [0.0, 1.0]  regularly spaced with Δz=0.0625
+
+julia> velocity_bcs = FieldBoundaryConditions(immersed=ValueBoundaryCondition(0));
+
+julia> model = NonhydrostaticModel(; grid, boundary_conditions=(u=velocity_bcs, v=velocity_bcs, w=velocity_bcs));
+
+julia> model.velocities.w.boundary_conditions.immersed
+ImmersedBoundaryCondition:
+├── west: ValueBoundaryCondition: 0
+├── east: ValueBoundaryCondition: 0
+├── south: ValueBoundaryCondition: 0
+├── north: ValueBoundaryCondition: 0
+├── bottom: Nothing
+└── top: Nothing
```

Comment:
Shouldn't these be also `ValueBoundaryCondition: 0`? Or at least the top BC (since it's a grid-fitted bottom)

Reply:
This is code produced by the doctest, so it here because that's default behavior!

As for what's going on --- the default boundary condition on velocity components that are normal to cell interfaces is `nothing`. (Here this is a vertical velocity.) As explained above, we cannot use `ValueBoundaryCondition(0)` for wall-normal velocity components.

Reply:
Ah, I missed the fact that this is a vertical velocity. Good point :+1: